scheduling_strawman.py

```diff
@@ -0,0 +1,92 @@
+"""
+The scheduler is a very useful tool. It requires the input of observing blocks
+and the time_range. Observing blocks consist of a target, a duration and constraints.
+"global" constraints can also be defined and can significantly speed up the scheduler.
+"""
+
+# a few packages are needed to set up the proper objects
+import numpy as np
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+from astropy.time import Time
+import astroplan
+from astroplan.constraints import *
+from astroplan import Observer, FixedTarget
+from astroplan.scheduling import *
+
+# create the time frame during which you observing
+start_time = Time.now()+4*u.hour
+end_time = start_time+24*u.hour
+
+# define the observer
+mro = Observer.at_site('mro')
+
+# make the targets (this can also use FixedTarget(coords) if you make a
+#                   `SkyCoord` object "coords")
+targets = [FixedTarget.from_name('vega'),
+           FixedTarget.from_name('Arcturus'),
+           FixedTarget.from_name('Altair'),
+           FixedTarget.from_name('Aldebaran'),
+           FixedTarget.from_name('Sirius'),
+           FixedTarget.from_name('Betelgeuse'),
+           FixedTarget.from_name('Rigel'),
+           FixedTarget.from_name('Castor')]
+target_group = [FixedTarget.from_name('Alcor'),
+                FixedTarget.from_name('M101')]
+
+priorities = {target.name: i for i, target in enumerate(targets)}
+
+# now `ObservingBlock`s and `BlockGroup`s need to be defined
+visit_duration = 55*u.minute
+blocks = []
+
+# an `ObservingBlock` contains the observation of one target for a duartion.
+
+# create a block for each target with the constraint that the airmass be less than 3
+# for that target during its OB and a non-boolean score (since constraints default to a boolean)
+# so that lower airmass gets a higher score.
+for target in targets:
+    OB = ObservingBlock(target, visit_duration, priority=priorities[target.name],
+                        constraints=[AirmassConstrant(max=3, boolean_score=False)]
```

Comment:
There should be an additional keyword argument `constraint_weights` which takes a weight for each constraint, so that you can choose which constraints matter most to you.